src/Common/src/System/IO/PathInternal.Windows.StringBuffer.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+
+namespace System.IO
+{
+    /// <summary>Contains internal path helpers that are shared between many projects.</summary>
+    internal static partial class PathInternal
+    {
+        /// <summary>
+        /// Returns true if the path uses the extended syntax (\\?\)
+        /// </summary>
+        internal static bool IsExtended(StringBuffer path)
+        {
+            return path != null && path.StartsWith(ExtendedPathPrefix);
```

Comment:
In what situation do we pass in a null path?  Should the null check just be in a Debug.Assert, or is that actually a valid case we take advantage of?